src/clj/rems/application/model.clj

```diff
@@ -18,82 +18,76 @@
   [application _event]
   application)
 
-(def ^:private draft-permissions
-  {:applicant [:application.command/save-draft
-               :application.command/submit
-               :application.command/close
-               :application.command/remove-member
-               :application.command/invite-member
-               :application.command/uninvite-member
-               :application.command/accept-licenses
-               :application.command/change-resources]
+(def ^:private submittable-application-commands
+  [:application.command/save-draft
+   :application.command/submit
+   :application.command/close
+   :application.command/remove-member
+   :application.command/invite-member
+   :application.command/uninvite-member
+   :application.command/accept-licenses
+   :application.command/change-resources])
+
+(def ^:private non-submittable-application-commands
+  [:application.command/remove-member
+   :application.command/uninvite-member
+   :application.command/accept-licenses])
+
+(def ^:private handler-all-commands
+  [:application.command/remark
+   :application.command/add-licenses
+   :application.command/add-member
+   :application.command/change-resources
+   :application.command/remove-member
+   :application.command/invite-member
+   :application.command/uninvite-member
+   :application.command/request-comment
+   :application.command/request-decision
+   :application.command/return
+   :application.command/approve
+   :application.command/reject])
+
+(def ^:private handler-returned-commands
+  (vec (remove #(= % :application.command/return) handler-all-commands)))
```

Comment:
handler-all-commands could be a set, which would simplify this code to just a disj

Reply:
A nice point, done. Although it adds some inconsistency between handler-all-commands and, e.g., submittable-application-commands.

Reply:
For consistency you could convert them all to sets.